packages/did/src/commonHelpers.ts

```diff
@@ -0,0 +1,184 @@
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Submodule to avoid circular dependencies. Re-exported under utils namespace.
+ */
+
+import {
+  DidKey,
+  DidServiceEndpoint,
+  EncryptionAlgorithms,
+  EncryptionKeyType,
+  NewDidKey,
+  SigningAlgorithms,
+  VerificationKeyType,
+} from '@kiltprotocol/types'
+import { SDKErrors } from '@kiltprotocol/utils'
+import { ApiPromise } from '@polkadot/api'
+import { u32 } from '@polkadot/types'
+import { stringToU8a } from '@polkadot/util'
+
+const signatureAlgForKeyType: Record<VerificationKeyType, SigningAlgorithms> = {
+  [VerificationKeyType.Ed25519]: SigningAlgorithms.Ed25519,
+  [VerificationKeyType.Sr25519]: SigningAlgorithms.Sr25519,
+  [VerificationKeyType.Ecdsa]: SigningAlgorithms.EcdsaSecp256k1,
+}
+export function getSigningAlgorithmForVerificationKeyType(
+  keyType: VerificationKeyType
+): SigningAlgorithms {
+  return signatureAlgForKeyType[keyType]
+}
+const keyTypeForSignatureAlg: Record<SigningAlgorithms, VerificationKeyType> = {
+  [SigningAlgorithms.Ed25519]: VerificationKeyType.Ed25519,
+  [SigningAlgorithms.Sr25519]: VerificationKeyType.Sr25519,
+  [SigningAlgorithms.EcdsaSecp256k1]: VerificationKeyType.Ecdsa,
+}
+export function getVerificationKeyTypeForSigningAlgorithm(
+  signatureAlg: SigningAlgorithms
+): VerificationKeyType {
+  return keyTypeForSignatureAlg[signatureAlg]
+}
+
+const encryptionAlgForKeyType: Record<EncryptionKeyType, EncryptionAlgorithms> =
+  {
+    [EncryptionKeyType.X25519]: EncryptionAlgorithms.NaclBox,
+  }
+export function getEncryptionAlgorithmForEncryptionKeyType(
+  keyType: EncryptionKeyType
+): EncryptionAlgorithms {
+  return encryptionAlgForKeyType[keyType]
+}
+const keyTypeForEncryptionAlg: Record<EncryptionAlgorithms, EncryptionKeyType> =
+  {
+    [EncryptionAlgorithms.NaclBox]: EncryptionKeyType.X25519,
+  }
+export function getEncryptionKeyTypeForEncryptionAlgorithm(
+  encryptionAlg: EncryptionAlgorithms
+): EncryptionKeyType {
+  return keyTypeForEncryptionAlg[encryptionAlg]
+}
+
+export function isVerificationKey(key: NewDidKey | DidKey): boolean {
+  return Object.values(VerificationKeyType).some((kt) => kt === key.type)
+}
+
+export function isEncryptionKey(key: NewDidKey | DidKey): boolean {
+  return Object.values(EncryptionKeyType).some((kt) => kt === key.type)
+}
+
+export function isUri(s: string): boolean {
+  try {
+    const url = new URL(s) // this actually accepts any URI but throws if it can't be parsed
+    return url.href === s || encodeURI(decodeURI(s)) === s // make sure our URI has not been converted implictly by URL
+  } catch {
+    return false
+  }
+}
+
+const UriFragmentRegex = /^[a-zA-Z0-9._~%+,;=*()'&$!@:/?-]+$/
+export function isUriFragment(s: string): boolean {
+  try {
+    return UriFragmentRegex.test(s) && !!decodeURIComponent(s)
+  } catch {
+    return false
+  }
+}
+
+export function checkServiceEndpointSyntax(
```

Comment:
What is the reason for returning the error and not throwing it?
Instead of returning an error I would maybe combine the string to list the problems and return a single error.

Reply:
Also a possible solution. I thought this is something you would want to do on a higher level (like I did it on `packages/did/src/DidBatcher/FullDidBuilder.ts`).